controllers/scopeinstance_controller.go

```diff
@@ -58,47 +67,166 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
-	// get 	the scope template
+	log.Log.Info("After getting ScopeInstance", "namespace", req.NamespacedName)
+
 	st := &operatorsv1.ScopeTemplate{}
 	if err := r.Client.Get(ctx, client.ObjectKey{Name: in.Spec.ScopeTemplateName}, st); err != nil {
+		// 4. Updating a ScopeInstance CR to reference a non-existent ScopeTemplate causes existing RBAC to be deleted.
+		// You can check here if RBAC present if yes then delete it
+		// delete RBAC and ScopeInstance both
+		log.Log.Info("throwing error check 4th condition scope template")
 		return ctrl.Result{}, err
 	}
 
 	log.Log.Info("ScopeTemplate found", "name", st.Name)
 
+	existingRoleBindingList := &rbacv1.RoleBindingList{}
+	existingClusterRoleBindingList := &rbacv1.ClusterRoleBindingList{}
+
+	// Get the existing ClusterRoleBinding
+	if err := r.Client.List(ctx, existingClusterRoleBindingList, client.MatchingLabels{
+		"labels": "scopeInstanceController",
```

Comment:
I'm sure that you know this, but this selector needs to be more unique. Let's add a TODO

```suggestion
// TODO(@laxmikantbpandhare): Make scopeInstance label unique.
		"labels": "scopeInstanceController",
```

Reply:
Updated label as `scopeInstance: operators.io.operator-framework/v1` Please let me know if that works.

Reply:
This isn't what I had in mind. What happens if there are two scopeInstances on cluster. The clusterRoleBindings returned will belong to both CRs.

Typically, we would create a label like `operators.coreos.io/scopeInstance: <Some Generated Key>`. It should be unique and reproducible per scopeInstance and [must be 63 characters or fewer](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) as it will fail label validation otherwise.

Reply:
Sure, I will look for this unique key generation process. and add like below as `/` does not work in labels
`operators.coreos.io-scopeInstance: <Some Generated Key>`